src/components/avatar/index.js

```diff
@@ -217,9 +219,11 @@ export default class Avatar extends BaseComponent {
   }
 
   renderRibbon() {
-    const {ribbonLabel, ribbonStyle, ribbonLabelStyle} = this.props;
+    const {ribbonLabel, ribbonStyle, ribbonLabelStyle, customRibbon} = this.props;
     if (ribbonLabel) {
-      return (
+      return customRibbon ? (
+        <View style={this.styles.customRibbon}>{customRibbon}</View>
```

Comment:
according to what u decide on the proptype of `customRibbon` you need to update this code. 
currently if someone passed a function, this code will not work